src/pyhs3/workspace.py

```
@@ -340,47 +340,119 @@ def _compute_observables(self) -> dict[str, tuple[float, float]]:
         Returns:
-            Model: The constructed model object.
+            :class:`~pyhs3.model.Model`: The constructed model.
         """
+        resolved_likelihood: Likelihood | None = None
+
+        if isinstance(target, Analysis):
+            # Derive everything from the analysis itself.
+            analysis_domain = target.domains[0]
+            if isinstance(analysis_domain, str):
+                msg = f"Analysis '{target.name}' domains must be FK-resolved before ws.model()"
+                raise RuntimeError(msg)
```

Ruff (0.15.12)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)